src/lib/state/workspace.svelte.ts

```diff
@@ -118,7 +118,12 @@ function createPortState() {
 	subscribe(async (port) => {
 		if (!port || reserved) return;
 		if (!port.readable || !port.writable) {
-			await port.open({ baudRate: 115200 });
+			try {
+				await port.open({ baudRate: 115200 });
+			} catch (e) {
+				onReady();
```

Comment:
It would probably be better if we would reject the ready promise instead of resolving this, as calling ready indicates to the rest of the code that it can start using the port, even though the opposite is true, rejecting this promise with a proper error message will assure that when opening the port fails, we will know that it was the opening of the port that failed and not some arbitrary read/write error later on

Reply:
Should be good now